project-koku/koku
# [Cost 2466] Add HCS automation into current Cost data pipeline

Fixes [COST-2466](https://issues.redhat.com/browse/COST-2466)

Changes proposed in this PR:
* add HCS to data pipeline, so that we are automatically processing HCS data daily


Testing Instructions:
1. Start koku from a clean run
 `make docker-up-min-trino`

3. Add `acct10001` to  hcs-data-processor Feature Toggle in unleash server
- http://localhost:4242/#/features/strategies/h
![Screen Shot 2022-04-20 at 8 28 39 PM](https://user-images.githubusercontent.com/57504257/164345961-7d2bcc3e-ec9a-4dac-ae79-bb4f91ee1e70.png)
4. look at minIO to make sure you see no "data" and no "hcs" under the koku-bucket
- http://localhost:9090/buckets/koku-bucket/browse
![Screen Shot 2022-04-20 at 8 33 38 PM](https://user-images.githubusercontent.com/57504257/164347008-f59cf8bc-6917-4010-b0db-a0e3be96e90a.png)
5. Create test customer
`make create-test-customer`
6.Load up test customer data for AWS(saves time)
`make load-test-customer-data test_source=AWS`
7.Once finished you can no w look at minIO again and you should see a data and hcs, sub directory under the koku-bucket
- http://localhost:9090/buckets/koku-bucket/browse
![Screen Shot 2022-04-20 at 8 34 17 PM](https://user-images.githubusercontent.com/57504257/164347879-b40f12b1-8544-498d-8e7f-cfea7f4e8731.png)

---
Additional Context:
You can traverse further down the hcs path and see that there are CSV files.
